sys/kern/exec.c

```diff
@@ -398,9 +398,8 @@ __noreturn void run_program(const char *path, char *const *argv,
                             char *const *envv) {
   proc_t *p = proc_self();
 
-  pgrp_enter(p, 1);
-
   assert(p != NULL);
+  assert(pgrp_enter(p, p->p_pid) == 0);
```

Comment:
It's always a bad idea to pass expression that have side effects to macros.
What if I decide to compile kernel with `NDEBUG` defined?
https://mimiker.ii.uni.wroc.pl/source/xref/mimiker/include/assert.h?r=fc854717#6

Reply:
Oh, right. Fixed.